src/motor_control/motor_control/PID_controller.py

```diff
@@ -1,170 +1,29 @@
-#PID Loop: only uses P and D term. I think integral term is not needed because, unlike cars and stuff, stopping the motors will cause
-# the sub to glide and not instantly stop which should reduce the problem of the sub getting infinitely close to the goal. I think it works but it is untested
-# and could have dumb errors. Kp and Kd values will need to be tuned.
-import rclpy
-from rclpy.node import Node
-import time
-from geometry_msgs.msg import Twist, Pose
-from std_msgs.msg import Bool
 import yaml
-import math
-
-KP = 0
-KD = 0
+import time
+gKP = 0
+gKD = 0
+gKI = 0
 
 with open('src/cfg/sub_properties.yaml') as f:
     file = yaml.safe_load(f)
-    KP = file['kp']
-    KD = file['kd']
-
-class PIDController(Node):
-
-    def __init__(self):
-        super().__init__('pid_controller')
-        self.goalpose_sub = self.create_subscription(Pose, 'goal_pose', self.goal_pose_callback, 10)
-        self.currentpose_sub = self.create_subscription(Pose, 'pose', self.current_pose_callback, 10)
-        self.auto_sub = self.create_subscription(Bool, 'autonamous_status', self.autonamous_status_callback, 10)
-        self.cmd_pub = self.create_publisher(Twist, 'sys_cmd_vel', 10)
-        
-        self.goal = Pose()
-        self.currentPosition = Pose()
-        self.deadzone = 0.1
+    gKP = file['kp']
+    gKD = file['kd']
+    gKI = file['ki']
+
+class PID():
+    # init with default values in YAML if values are not passed
+    def __init__(self, KP = gKP, KD = gKP, KI = gKI):
+        self.KP = KP
+        self.KD = KD
+        self.KI = KI
+        self.prev_error = 0
+        self.integral = 0
+        self.prev_time = 0
+    def calculateOutput(self, state, goal):
+        error = goal - state
+        integral = self.integral + error
```

Comment:
Shouldn't the integral term also be scaled by time?
`integral = self.integral + (error * (time.time() - self.prev_time))`

Reply:
For the velocity implememtation yes ... right?